### DISCUSSION TITLE
Multiple selectors for style class syntax

### DISCUSSION THREAD

**Discussion by wgraaf** [AvaloniaUI/Avalonia#11698] [Q&A] [answered]
I would like to create a style class for a transparent button on a a pointerover, as well as default.

This style works for pointerover:

    <Style Selector="Button:pointerover.transparent /template/ ContentPresenter">
      <Setter Property="Background" Value="Transparent"/>
    </Style>

I can set Background="Transparent" in the button element, but according to  https://github.com/AvaloniaUI/Avalonia/issues/1743
it might be possible to add a 'Button' selector to the style class using a comma as separator.

However, all attempts to do so result in axaml errors.
What would the syntax be to combine both 'Button' and 'Button:pointerover' into the selector?

**Comment by robloo** [comment]
I've often seen multiple selectors on different lines, although, not with classes. So I'm not 100% sure this will work.

```xaml
<Style Selector="Button.transparent /template/ ContentPresenter,
                 Button:pointerover.transparent /template/ ContentPresenter">
  <Setter Property="Background" Value="Transparent"/>
</Style>
```

Within a ControlTheme something like this should also work:

https://github.com/AvaloniaUI/Avalonia/blob/dc9b47b9bef586b404e04372bbd4bb47cd14f3fb/src/Avalonia.Controls.ColorPicker/Themes/Fluent/ColorSpectrum.xaml#L86-L90

Generally, for 11.0+, I would recommend just switching to a full control theme to create a "clear" button theme. That's what is done in some apps on my end. It make more sense when you start customizing all the pseudoclass states and you can also use `BasedOn` so you don't need to duplicate too much.

**Comment by wgraaf** [comment, marked as answer]
Thanks!
The trick was to spell out the selector twice, separated by a comma.
The selectors can be on the same line or on separate lines:

    <Style Selector="Button.transparent /template/ ContentPresenter, Button:pointerover.transparent /template/ ContentPresenter">
      <Setter Property="Background" Value="Transparent"/>
    </Style>

or

    <Style Selector="Button.transparent /template/ ContentPresenter, 
                     Button:pointerover.transparent /template/ ContentPresenter">
      <Setter Property="Background" Value="Transparent"/>
    </Style>

both work.
